sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -377,8 +437,9 @@ class StreamExecution(
     val batchTime = (System.nanoTime() - startTime).toDouble / 1000000
     logInfo(s"Completed up to $availableOffsets in ${batchTime}ms")
     // Update committed offsets.
-    committedOffsets ++= availableOffsets
-    postEvent(new QueryProgress(this.toInfo))
+    statusLock.synchronized {
```

Comment:
Why? committed offsets is being updated, and can get updated in the middle of the query status generation.